pkg/dockermanager/dockermanager.go

```diff
@@ -0,0 +1,197 @@
+package dockermanager
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+
+	"github.com/ipfs/testground/pkg/logging"
+)
+
+// Manager is a convenient wrapper around the docker client.
+type Manager struct {
+	logging.Logging
+	*client.Client
+}
+
+// NewManager connects to the local docker instance and provides a convenient
+// handle for managing containers.
+func NewManager() (*Manager, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return nil, err
+	}
+
+	return &Manager{
+		Logging: logging.NewLogging(logging.S().With("host", cli.DaemonHost()).Desugar()),
+		Client:  cli,
+	}, nil
+}
+
+// Close closes the docker client.
+func (dm *Manager) Close() error {
+	return dm.Client.Close()
+}
+
+// Container is a convenient handle for a docker container.
+type Container struct {
+	logging.Logging
+	ID      string
+	Manager *Manager
+}
+
+// Inspect inspects this docker container.
+func (dc *Container) Inspect(ctx context.Context) (types.ContainerJSON, error) {
+	return dc.Manager.ContainerInspect(ctx, dc.ID)
+}
+
+// IsOnline returns whether or not the container is online.
+func (dc *Container) IsOnline(ctx context.Context) (bool, error) {
+	info, err := dc.Inspect(ctx)
+	if err != nil {
+		if client.IsErrNotFound(err) {
+			err = nil
+		}
+		return false, err
+	}
+	switch info.ContainerJSONBase.State.Status {
+	case "running", "paused":
```

Comment:
You can pause (SIGSTOP?) containers while they're running. When you resume them, they'll start where you left off.

I have this logic here in case someone wants to pause then resume a test, or maybe pause specific machines (?) for some reason.